# Problem 1

```
#!/bin/bash

module avail fastqc
module load fastqc/0.11.9

ls /pub/mbarcen1/EE283/DNAseq/rawdata

fastqc /pub/mbarcen1/EE283/DNAseq/rawdata/ADL06_1_1.fq.gz /pub/mbarcen1/EE283/DNAseq/rawdata/ADL06_1_2.fq.gz

module load trimmomatic/0.39
module load java/1.8.0

java -jar /opt/apps/trimmomatic/0.39/trimmomatic-0.39.jar PE -threads 4 -phred33 \
/pub/mbarcen1/EE283/DNAseq/rawdata/ADL06_1_1.fq.gz /pub/mbarcen1/EE283/DNAseq/rawdata/ADL06_1_2.fq.gz \
/pub/mbarcen1/EE283/DNAseq/output/ADL06_1_1_paired.fq.gz /pub/mbarcen1/EE283/DNAseq/output/ADL06_1_1_unpaired.fq.gz \
/pub/mbarcen1/EE283/DNAseq/output/ADL06_1_2_paired.fq.gz /pub/mbarcen1/EE283/DNAseq/output/ADL06_1_2_unpaired.fq.gz \
ILLUMINACLIP:/opt/apps/trimmomatic/0.39/adapters/NexteraPE-PE.fa:2:30:10 LEADING:3 SLIDINGWINDOW:4:15 MINLEN:30


```

# Problem 2
```
#!/bin/bash

module avail fastqc
module load fastqc/0.11.9

ls /pub/mbarcen1/EE283/ATACseq/rawdata

fastqc inputfile1 inputfile2
```

# Problem 3

## Acquire reference
```
SourceDir="/data/class/ecoevo283/public/Bioinformatics_Course/ref"
DestDir="/pub/$USER/EE283/ref"

ln -s $SourceDir/dmel-all-chromosome-r6.13.fasta $DestDir/dmel-all-chromosome-r6.13.fasta
ln -s $SourceDir/dm6.fa.gz $DestDir/dm6.fa.gz
ln -s $SourceDir/dm6.ncbiRefSeq.gtf.gz $DestDir/dm6.ncbiRefSeq.gtf.gz
ln -s $SourceDir/dmel-all-r6.13.gtf $DestDir/dmel-all-r6.13.gtf

```

## Create the sequence dictionary

```
module load java/1.8.0
module load picard-tools/2.27.1
module load bwa/0.7.8
module load samtools/1.10

ref="/pub/$USER/EE283/ref/dmel-all-chromosome-r6.13.fasta"
bwa index $ref
samtools faidx $ref
# what is wrong with the command below?
java -jar /opt/apps/picard-tools/2.27.1/picard.jar \
CreateSequenceDictionary R=$ref O=/pub/$USER/EE283/ref/dm6.dict
```
# Problem 4

## Test out a small script on small scale data derived from main data

```
# Define paths
ref="/pub/$USER/EE283/ref/dmel-all-chromosome-r6.13.fasta"
rawDataDir="/pub/$USER/EE283/DNAseq/rawdata"
outputDir="/pub/$USER/EE283/DNAseq/test_output"

# Ensure output directory exists
mkdir -p $outputDir

# Create test data (1 million reads, 4 million lines)
zcat $rawDataDir/ADL06_1_1.fq.gz | head -n 4000000 | gzip -c > $outputDir/test_1.fq.gz
zcat $rawDataDir/ADL06_1_2.fq.gz | head -n 4000000 | gzip -c > $outputDir/test_2.fq.gz
```

Next create a script to run this test file. 

### Create the shell script to be submitted via SBATCH 

```
#!/bin/bash
#SBATCH --job-name=test    ## Name of the job.
#SBATCH -A CLASS-ECOEVO283       ## account to charge
#SBATCH -p standard        ## partition/queue name
#SBATCH --array=1-20   ## number of tasks to launch (wc -l prefixes.txt)
#SBATCH --cpus-per-task=2    ## number of cores the job needs 
#SBATCH --error=hw3_p3_ee283.%J.err
#SBATCH --output=hw3_p3_ee283.%J.out

module load java/1.8.0
module load picard-tools/2.27.1

ref="/pub/$USER/EE283/ref/dmel-all-chromosome-r6.13.fasta"
bwa index $ref
samtools faidx $ref
# what is wrong with the command below?
java -jar /opt/apps/picard-tools/2.27.1/picard.jar \ CreateSequenceDictionary R=$ref O=dm6.dict

sleep 2m	# wait 2 minutes
```

## For DNAseq 
Check the directory for file: DNAseq/scripts/hw3_ee283_dnaseq_align.sh

## For ATACseq 
Check the directory for file: ATACseq/scripts/hw3_ee283_atacseq_align.sh

# Problem 5

## For RNAseq

Check the directory for files: for indexing RNAseq/scripts/hw_ee283_rnaseq.index.sh and for alignment RNAseq/scripts/hw3_ee283_rnaseq_align.sh

