# Problem 1

##  Write, and run the script to call SNPs in the DNAseq dataset.

[DNAseq/scripts/hw5_ee2823_p1.sh](https://github.com/barcenasmanuel/EE283/blob/main/DNAseq/scripts/hw5_ee2823_p1.sh)

# Problem 2

## Filter your SNPs calls.  Try to put some sort of plot in your git of what the loss of SNPs is at each filtering step (what is an easy way to estimate the # of SNPs in a vcf?).  Filtering does come at a price.

In progress

# Problem 3

## Try to represent the filtered biallelic SNPs from the first 1Mb of the X chromosome graphically.  The idea is to have a graphic that is almost like a big matrix where rows are the four different genotypes you have processed, and columns are different SNPs (that have passed filters). Each cell is colored green/yellow/red where green is ref/ref, yellow is ref/alt, red is alt/alt.  

Hints:

a) check out vcftools (module avail) and its output option called “--012”.
b) In R the image command with switches col=c(“green”,”yellow”,”red”) and breaks=c(-0.5,0.5,1.5,2.5)
c) you may have to rotate the matrix passed to the image command

This is a little tricky, but allows for a complex visualization of SNP data.

In progress


# Problem 4

## The FOUR strains you are looking at are mostly isogenic (as they are highly inbred).  Thus most SNP calls should be 0/0 or 1/1.  Repeat #3, but only consider SNPs where two of the strains are 0/0 and the other two 1/1. That is a SNP at a frequency of 50%.

## This is your chance to use awk on the output of the “--012” command, first remove rows that have a “1” in any of the 4 genotype fields, as a "1" indicates the call is heterozygous in that line.  Then only consider rows where the sum of the last 4 fields is 4.  That is, two strains are "2" and two strains are "0". This problem is a little harder, but you often want to filter on allele frequency and/or genotypes in subsets of strains.  It is more important to solve if you want to do population or transmission genetics.


