# Problem 1

## Do QC on 6 of your ATACseq bam files.  Try to include the different tissue types.

For this portion, we realigned the bam files to UCSC genome. We used the chromX subset since we wanted to get an idea for the processing.

For the code to implement the following images look at [ATACseq/scripts/.ipynb](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/scripts/HW6_ee283_p1.ipynb)

### Looking at distribution of fragement lengths

A4_ED_2.chrX

![A4_ED_2.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/fragment_size_plots/A4_ED_2.chrX_fragsize.png)

A4_WD_2.chrX

![A4_WD_2.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/fragment_size_plots/A4_WD_2.chrX_fragsize.png)

A5_ED_3.chrX

![A5_ED_3.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/fragment_size_plots/A5_ED_3.chrX_fragsize.png)

A5_WD_3.chrX

![A5_WD_3.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/fragment_size_plots/A5_WD_3.chrX_fragsize.png)

A6_ED_1.chrX

![A6_ED_1.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/fragment_size_plots/A6_ED_1.chrX_fragsize.png)

A6_WD_1.chrX

![A6_WD_1.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/fragment_size_plots/A6_WD_1.chrX_fragsize.png)

### Looking at library complexity

A4_ED_2.chrX

![A4_ED_2.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/library_complexity_plots/A4_ED_2.chrX_libcomplexity.png)

A4_WD_2.chrX

![A4_WD_2.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/library_complexity_plots/A4_WD_2.chrX_libcomplexity.png)

A5_ED_3.chrX

![A5_ED_3.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/library_complexity_plots/A5_ED_3.chrX_libcomplexity.png)

A5_WD_3.chrX

![A5_WD_3.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/library_complexity_plots/A5_WD_3.chrX_libcomplexity.png)

A6_ED_1.chrX

![A6_ED_1.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/library_complexity_plots/A6_ED_1.chrX_libcomplexity.png)

A6_WD_1.chrX

![A6_WD_1.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/library_complexity_plots/A6_WD_1.chrX_libcomplexity.png)

### Looking at distribution of reads mapping near TSSs

A4_ED_2.chrX

![A4_ED_2.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/heatmap/A4_ED_2.chrX_heatmap.png)

A4_WD_2.chrX

![A4_WD_2.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/heatmap/A4_WD_2.chrX_heatmap.png)

A5_ED_3.chrX

![A5_ED_3.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/heatmap/A5_ED_3.chrX_heatmap.png)

A5_WD_3.chrX

![A5_WD_3.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/heatmap/A5_WD_3.chrX_heatmap.png)

A6_ED_1.chrX

![A6_ED_1.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/heatmap/A6_ED_1.chrX_heatmap.png)
A6_WD_1.chrX

![A6_WD_1.chrX](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/output/ucsc/sub_processed/heatmap/A6_WD_1.chrX_heatmap.png)

# Problem 2

## Generate a broad peaks bigwig for one of the 4 tissues (by combining samples within that tissue and running MACS2).  Try serving it up from cyverse (https://cyverse.org/; the free tier should be OK for this course), or some other option you have access to (e.g, your lab has a webserver, etc).

Could not figure out how to host file, however I did run code to make bigwig files:

To generate first step: [ATACseq/scripts/hw6_p2_ucsc.sh](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/scripts/hw6_ee283_p2_ucsc.sh)

To generate bigwig: [ATACseq/scripts/hw6_p2_ucsc_bw.sh](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/scripts/hw6_ee283_p2_ucsc_bw.sh)
# Problem 3 

## Some people do not like the way MAC2/bigwig displays its broad peaks file.  So I propose rolling our own.  Start with a sorted indexed bam file, perhaps made by combining all the different genotypes within a tissue (or if you wish compare two genotypes).  Then we will just calculate read-span coverage throughout the genome.  I will normalize by the number of reads in the bam.

Could not figure out how to host file, however I did run code to make bigwig files:

[ATACseq/scripts/hw6_p3_ucsc_bw.sh](https://github.com/barcenasmanuel/EE283/blob/main/ATACseq/scripts/hw6_ee283_p3_ucsc_bw.sh)

# Problem 4 

## You could also explore the poorly documented -pc switch of genomeCoverageBed above.  What is this switch doing (is it clear from the documentation).  The “bamCoverage” tool from deeptools (of week 4) is perhaps more transparent, and can output directly to bigwig, and has some automatic switches for normalization, and has some extra switches that could be useful.  See if you can get additional coverage tracks using this tool (and check they are the same!).

This is skipped

