# HW7- RNAseq

# 1. Do the same subsetting of the data I do early in this chapter using base R commands.

Create the symlink:
```
ln -s /data/class/ecoevo283/public/Bioinformatics_Course/RNAseq/RNAseq384_SampleCoding.txt /pub/mbarcen1/EE283/hw7/RNAseq.samcode.txt

```
Code in base R without tidyverse:
```
# Read the data
mytab <- read.delim("RNAseq.samcode.txt", header = TRUE, sep = "\t", stringsAsFactors = FALSE)

# Display the data
print(mytab)

# Select columns
mytab2 <- mytab[, c("RILcode", "TissueCode", "Replicate", "FullSampleName")]

# Display frequency tables
print(table(mytab2$RILcode))
print(table(mytab2$TissueCode))
print(table(mytab2$Replicate))

# Filter the data
ril_filter <- c(21148, 21286, 22162, 21297, 21029, 22052, 22031, 21293, 22378, 22390)
tissue_filter <- c("B", "E")
replicate_filter <- "0"

mytab2 <- mytab2[mytab2$RILcode %in% ril_filter & 
                 mytab2$TissueCode %in% tissue_filter & 
                 mytab2$Replicate == replicate_filter, ]

# Write file paths to a text file
file_paths <- paste0("RNAseq/bam/", mytab2$FullSampleName, ".bam")
writeLines(file_paths, "shortRNAseq.names.txt")

# Write filtered data to a tab-separated file
write.table(mytab2, "shortRNAseq.txt", sep = "\t", row.names = FALSE, quote = FALSE)

```

We get the bam files:

```
RNAseq/bam/21148B0.bam
RNAseq/bam/21286B0.bam
RNAseq/bam/22162B0.bam
RNAseq/bam/21297B0.bam
RNAseq/bam/21029B0.bam
RNAseq/bam/22052B0.bam
RNAseq/bam/22031B0.bam
RNAseq/bam/21293B0.bam
RNAseq/bam/22378B0.bam
RNAseq/bam/22390B0.bam
RNAseq/bam/21148E0.bam
RNAseq/bam/21286E0.bam
RNAseq/bam/22162E0.bam
RNAseq/bam/21297E0.bam
RNAseq/bam/21029E0.bam
RNAseq/bam/22052E0.bam
RNAseq/bam/22031E0.bam
RNAseq/bam/21293E0.bam
RNAseq/bam/22378E0.bam
RNAseq/bam/22390E0.bam

```

as well as the table:

```
RILcode	TissueCode	Replicate	FullSampleName
21148	B	0	21148B0
21286	B	0	21286B0
22162	B	0	22162B0
21297	B	0	21297B0
21029	B	0	21029B0
22052	B	0	22052B0
22031	B	0	22031B0
21293	B	0	21293B0
22378	B	0	22378B0
22390	B	0	22390B0
21148	E	0	21148E0
21286	E	0	21286E0
22162	E	0	22162E0
21297	E	0	21297E0
21029	E	0	21029E0
22052	E	0	22052E0
22031	E	0	22031E0
21293	E	0	21293E0
22378	E	0	22378E0
22390	E	0	22390E0
```

# 2.  Run DESEQ for your data.  If you wish you could consider different samples.

Readjusted script above to create filenames to align with my data:

```
RNAseq/output/21148_B_0.sorted.bam
RNAseq/output/21286_B_0.sorted.bam
RNAseq/output/22162_B_0.sorted.bam
RNAseq/output/21297_B_0.sorted.bam
RNAseq/output/21029_B_0.sorted.bam
RNAseq/output/22052_B_0.sorted.bam
RNAseq/output/22031_B_0.sorted.bam
RNAseq/output/21293_B_0.sorted.bam
RNAseq/output/22378_B_0.sorted.bam
RNAseq/output/22390_B_0.sorted.bam
RNAseq/output/21148_E_0.sorted.bam
RNAseq/output/21286_E_0.sorted.bam
RNAseq/output/22162_E_0.sorted.bam
RNAseq/output/21297_E_0.sorted.bam
RNAseq/output/21029_E_0.sorted.bam
RNAseq/output/22052_E_0.sorted.bam
RNAseq/output/22031_E_0.sorted.bam
RNAseq/output/21293_E_0.sorted.bam
RNAseq/output/22378_E_0.sorted.bam
RNAseq/output/22390_E_0.sorted.bam
```

Ran the following script to sbatch before running DESEQ on data:

```
#!/bin/bash
#SBATCH --job-name=featureCounts
#SBATCH -A CLASS-ECOEVO283
#SBATCH -p standard
#SBATCH --cpus-per-task=8
#SBATCH --mem=16G
#SBATCH --time=04:00:00
#SBATCH --output=logs/featureCounts.out
#SBATCH --error=logs/featureCounts.err

# Load the required module
module load subread/2.0.3

# Set the paths
gtf="/pub/mbarcen1/EE283/ref/dmel-all-r6.13.gtf"
output_dir="/pub/mbarcen1/EE283/RNAseq/counts"

# Create output directory if it doesn't exist
mkdir -p $output_dir

# Create a space-delimited list of BAM files
myfile=$(cat /pub/mbarcen1/EE283/hw7/shortRNAseq.names.txt | tr "\n" " ")

# Run featureCounts
featureCounts -p -T 8 -t exon -g gene_id -Q 30 -F GTF -a $gtf -o $output_dir/fly_counts.txt $myfile

echo "featureCounts completed. Output saved to $output_dir/fly_counts.txt"
```

For code please see [RNAseq/scripts/hw7_ee283_p2_2.ipynb](https://github.com/barcenasmanuel/EE283/blob/main/RNAseq/scripts/hw7_ee283_p2_2.ipynb)

# 3.  Create a volcano plot of your data.  These are popular plots you often see in publications.

For code please see [RNAseq/scripts/hw7_ee283_p2_2.ipynb](https://github.com/barcenasmanuel/EE283/blob/main/RNAseq/scripts/hw7_ee283_p2_2.ipynb)

