# Problem 1
## Use samtools and the gDNA seq bam files to extract a 120kb region of the genome (chrX from 1,880,000 to 2,000,000) from strain “A4”. Make sure you only include reads that have mapq > 30.  Also do this for a control strain (A5).

Here look at [DNAseq/scripts/hw4_ee2823_dnaseq_regions.sh](https://github.com/barcenasmanuel/EE283/blob/main/DNAseq/scripts/hw4_ee283_dnaseq_regions.sh)

# Problem 2 **(Extra problem if time permits)**
## Repeat the exercise of step #1 but instead extract reads mapping to different chromosomes whose mate maps to this region.  This means you cannot filter on quality score.  Extract them as fasta.  Then try to assemble using spades.  And figure out what is going on using blast.

## Can you draw a picture of what is going on in this region in A4?  What is a problem with totally trusting the output of a pipeline like GATK for identifying *ALL* variation?

# Problem 3
## Install deeptools as a conda environment: Using deeptools and the bam files of problem 1, plot the per bp read coverage for your regions for the two samples (normalize using RPKM).  Try to visualize A4 relative to A5 (especially around 1,904,042).  

Here look at [DNAseq/scripts/hw4_plotcoverage_dnaseq.sh](https://github.com/barcenasmanuel/EE283/blob/main/DNAseq/scripts/hw4_plotcoverage_dnaseq.sh)

# Problem 4
## Repeat the process but plot the fragment coverage (--extendReads).  Try to visualize A4 vs. A5 (especially around 1,904,042).

# Problem 5 (Extra problem if time permits)
## “The answer” can be found in the Santa Cruz Genome Browser for this region, where we have assembled each of these genomes.

[http://goo.gl/LLpoNH]

